src/data/actions/courseInfo.test.js

```diff
@@ -187,6 +188,43 @@ describe('courseInfo edit course actions', () => {
     const expectedAction = { type: types.CLEAR_COURSE_SAVED };
     expect(clearCourseSaved()).toEqual(expectedAction);
   });
+
+  it('form value update after save is called with passed change method', () => {
+    const changeMock = jest.fn();
+    const store = mockStore();
+    const formValues = {
+      geoLocationLng: '45.0000',
+      geoLocationLat: '50.0000',
+      url_slug: 'test_slug',
+      tags: ['tag1', 'tag2'],
+      in_year_value: {
+        per_lead_usa: 10,
+        per_lead_international: 20,
+        per_click_usa: 25,
+        per_click_international: 30,
+      },
+      imageSrc: 'http://updated.image.src/woo.small',
+      course_runs: [
+        {
+          status: 'published',
+          transcript_languages: ['en-us'],
+        },
+      ],
+    };
+    store.dispatch(updateFormValuesAfterSave(changeMock, formValues, formValues));
+
+    expect(changeMock).toHaveBeenNthCalledWith(1, 'imageSrc', 'http://updated.image.src/woo.small');
+    expect(changeMock).toHaveBeenNthCalledWith(2, 'geoLocationLat', '50.0000');
+    expect(changeMock).toHaveBeenNthCalledWith(3, 'geoLocationLng', '45.0000');
+    expect(changeMock).toHaveBeenNthCalledWith(4, 'tags', ['tag1', 'tag2']);
+    expect(changeMock).toHaveBeenNthCalledWith(5, 'url_slug', 'test_slug');
+    expect(changeMock).toHaveBeenNthCalledWith(6, 'in_year_value.per_lead_usa', 10);
+    expect(changeMock).toHaveBeenNthCalledWith(7, 'in_year_value.per_lead_international', 20);
+    expect(changeMock).toHaveBeenNthCalledWith(8, 'in_year_value.per_click_usa', 25);
+    expect(changeMock).toHaveBeenNthCalledWith(9, 'in_year_value.per_click_international', 30);
+    expect(changeMock).toHaveBeenNthCalledWith(10, 'course_runs[0].status', 'published');
+    expect(changeMock).toHaveBeenNthCalledWith(11, 'course_runs[0].transcript_languages', ['en-us']);
+  });
```

Comment:
nit: not a blocker but this can be reduced to a for-loop iterating over `formValues`.

Reply:
It required a JSON to map fields -> values and I was being a bit lazy with it, so I went with this approach.